log, camera and scoped storage

mmattel opened the pull request:
Fixes: https://github.com/owncloud/docs-client-android/issues/17 (Functionality updates)

Adding:

* Scoped Storage
* Logs view redesign
* Camera uploads when charging

Backport to 2.19

@jesmrec fyi

EParzefall commented on modules/ROOT/pages/accounts.adoc:
```diff
@@ -33,7 +33,7 @@ This action also logs you out of the server and deletes the database with the li
 image:accounts/android-remove-account-confirmation.jpg[{app-name}: Confirm removal of user account, width=40%,pdfwidth=40%]
 
-NOTE: There is no logout function that both logs out of your account *and* leaves all your files on your Android device.
+NOTE: When removing an account, the files related to that account are removed automatically. This is also true when uninstalling the app which removes all allcunts with their data.
```
```suggestion
NOTE: When removing an account, the files related to that account are removed automatically. This is also true when uninstalling the app which removes all accounts with their data.
```

EParzefall commented on modules/ROOT/pages/connecting.adoc:
```diff
@@ -1,15 +1,27 @@
 
+== Physical Location of User Files
+
+Compared to earlier versions of this app (< 2.19), the location for files created or managed by this app is the so called *Scoped Storage*. Scoped Storage is, compared to Shared Storage, protected and part of the app. It is located on the devices memory and limited by its capacity. Files are now safe and private and can only be accessed by other apps that use the {document-provider-url}[Documents ...
```
```suggestion
Compared to earlier versions of this app (< 2.19), the location for files created or managed by this app is called *Scoped Storage*. Scoped Storage is, in contrast to Shared Storage, protected and part of the app. It is saved in the device's memory and limited by its capacity. Files are now safe and private and can only be accessed by other apps that use the {document-provider-url}[Documents Provider]. Internal tests have successfully passed using more than 50GB of Scoped Storage without any issues.
```

mmattel replied:
committing, will adopt the memory -> "storage" part later

EParzefall commented on modules/ROOT/pages/connecting.adoc:
```diff
@@ -1,15 +1,27 @@
+
+When a connection is removed or the app is uninstalled, all files are removed for that connection or the whole app by design - securing data privacy.
+
+NOTE: Consider enough _internal storage capacity_ for your files being synced by the app. As a rule of thumb, check the capacity necessary via the owncloud web access first before enabling a sync for a particular mount and or path using offline access for files.
```
```suggestion
NOTE: Make sure there's enough _internal storage capacity_ for your files to be synced by the app. As a rule of thumb, check the capacity necessary via the ownCloud web access first before enabling a sync for a particular mount and or path using offline access for files.
```

EParzefall commented on modules/ROOT/pages/faq.adoc:
```diff
@@ -34,13 +34,11 @@ Please contact your ownCloud administrator for more details.
-Local changes get upload when you add files inside the app, or a 3rd party app sends a file to the ownCloud app, or a 3rd party app adds/edits files via the document provider.
-
-Local changes won't be uploaded when 3rd party apps directly access the file system. Android will prevent direct file system access in the near future (Scoped Storage)
+Local changes get recognized and upload when you add files inside the app, or a 3rd party app sends a file to the ownCloud app, or a 3rd party app adds/edits files via the document provider.
```
```suggestion
Local changes get recognized and uploaded when you add files inside the app, or a 3rd party app sends a file to the ownCloud app, or a 3rd party app adds/edits files via the document provider.
```

EParzefall commented on modules/ROOT/pages/installation.adoc:
```diff
@@ -26,6 +26,18 @@ When you download your {app-name} from the Google Play store, it will be upgrade
+====
+When upgrading from a release lower than 2.19, the app is using Scoped Storage instead of Shared Storage. See xref:connecting.adoc#physical-location-of-user-files[Physical Location of User Files] for more details.
+
+The upgrade process takes care transferring all files one by one from Shared Storage to Scoped Storage deleting them from Shared Storage post success for each file. The transition does not require user interaction, runs automatically and is neccessary only once.
```
```suggestion
The upgrade process takes care of transferring all files one by one from Shared Storage to Scoped Storage and deleting them from Shared Storage after a successful move of each file. The transition does not require user interaction, runs automatically and only needs to be performed once.
```

EParzefall commented on modules/ROOT/pages/installation.adoc:
```diff
@@ -26,6 +26,18 @@ When you download your {app-name} from the Google Play store, it will be upgrade
+
+[WARNING]
+====
+Do not power off you device while an upgrade runs the migration process from Shared Storage to Scoped Storage. In case this happens, even accidentially, ownCloud recommends to disconnect all connections and manually delete all Shared Storage data manually. When finished, re-setup all connections which are then automatically using Scoped Storage.
```
```suggestion
Do not power off you device while an upgrade runs the migration process from Shared Storage to Scoped Storage. If it happens by accident, ownCloud recommends to disconnect all connections and manually delete all Shared Storage data. When finished, set up all connections once more, which are then automatically using Scoped Storage.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -15,15 +15,22 @@ image:settings/android-settings-page.png[The Settings Screen in the {app-name},
 == Camera Uploads
 
-If you take photos or create videos with your Android device, they can be automatically uploaded to your ownCloud server. To enable this, under _"Camera uploads"_ tap one or both of _Picture uploads_ or _Video uploads_.
+If you take photos or create videos with your Android device, they can be automatically uploaded to your ownCloud server. To enable this, under _"Camera uploads"_ tap one or both of _Picture uploads_ or _Video uploads_. The following image shows the settings for the picture upload. The same settings are available on a separate screen for video uploads.
```
```suggestion
If you take photos or create videos with your Android device, they can be automatically uploaded to your ownCloud server. To enable this, under _"Camera uploads"_ tap _Picture uploads_ or _Video uploads_ or both. The following image shows the settings for the picture upload. The same settings are available on a separate screen for video uploads.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -15,15 +15,22 @@ image:settings/android-settings-page.png[The Settings Screen in the {app-name},
+NOTE: Be aware that if you used the earlier _Instant Uploads_ feature, you will lose the configuration and have to enable the "Camera uploads" feature if you want to use it, since it needs to be initialized and configured properly.
 
-If you’re concerned about mobile data usage, or have an account with limited data available, you can limit uploading to only when a WiFi is in use. This option is visible once you've enabled the respective option. For photos tap *"Upload pictures via wifi only"*. For videos tap *"Upload videos via wifi only"*.
+If you’re concerned about mobile data usage, or have an account with limited data available, you can limit uploading to only when a WiFi is in use. This option is visible once you've enabled the respective option.
```
```suggestion
If you’re concerned about mobile data usage or have an account with limited data available, you can limit uploading to only when a WiFi is in use.
```

EParzefall replied:
I can't picture how to enable an invisible option. Am I missing something here?

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -15,15 +15,22 @@ image:settings/android-settings-page.png[The Settings Screen in the {app-name},
+* For photos tap *Upload pictures via wifi only*
+* For videos tap *Upload videos via wifi only*
+
+In addition, you can select to upload only when charging the device avoiding battery drain.
```
```suggestion
In addition, you can select to upload only when charging the device to avoid battery drain.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -15,15 +15,22 @@ image:settings/android-settings-page.png[The Settings Screen in the {app-name},
+
+In addition, you can select to upload only when charging the device avoiding battery drain.
+
+NOTE: Selected conditional uploads are meant in the way, if both are selected for a type, both requirements must be fulfilled that uploads are processed. 
```
```suggestion
NOTE: If both conditional uploads are selected for a type, both requirements must be fulfilled for uploads to happen. 
```

phil-davis replied:
```suggestion
NOTE: All selected uploading conditions must be met for uploads to happen.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -15,15 +15,22 @@ image:settings/android-settings-page.png[The Settings Screen in the {app-name},
+
+NOTE: Selected conditional uploads are meant in the way, if both are selected for a type, both requirements must be fulfilled that uploads are processed. 
 
 By default, photos and videos are uploaded to a directory called `/CameraUpload`. However, you also have the option to use an existing directory, or to create a new one. To change the upload location, tap on _Picture upload path_ under photos or _Video upload_ path under videos, and choose one of the folders displayed.
```
```suggestion
By default, photos and videos are uploaded to a directory called `/CameraUpload`. However, you also have the option to use an existing directory or to create a new one. To change the upload location, tap on _Picture upload path_ under photos or _Video upload_ path under videos and choose one of the folders displayed.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -37,11 +44,41 @@ image:settings/allow-light-filtering-apps.png[Allow light filtering apps, width=
-image:logs/show-logs.png[View logs in the {app-name}, width=40%,pdfwidth=40%]
+If you detect a problem or a bug in the {app-name}, you can send us logs about the problem. You will find the list of logs in the Settings view, after enabling it. To enable the logs, you have to _tap five times on the version and build information_, available under menu:Settings[ownCloud for Android] .
+
+When enabled, inside the log section of the Android owncloud app, options to share and delete logs are available.
```
```suggestion
When enabled, inside the log section of the Android ownCloud app, options to share and delete logs are available.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -37,11 +44,41 @@ image:settings/allow-light-filtering-apps.png[Allow light filtering apps, width=
+
+When enabled, inside the log section of the Android owncloud app, options to share and delete logs are available.
+
+You can open a log for viewing/searching content by tapping on it. As a prerquisiste, an app must be installed capable reading/searching in log files. ownCloud does not provide a built-in log reader anymore as there are many professional apps available.
```
```suggestion
You can open a log for viewing content by tapping on it. As a prerequisite, an app capable of reading and searching in log files must be installed. ownCloud does not provide a built-in log reader anymore as there are many professional apps available.
```

phil-davis replied:
```suggestion
You can open a log for viewing/searching content by tapping on it. As a prerequisite, an app must be installed that is capable of reading/searching in log files. ownCloud does not provide a built-in log reader anymore as there are many professional apps available.
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -37,11 +44,41 @@ image:settings/allow-light-filtering-apps.png[Allow light filtering apps, width=
+
+image:logs/show-logs-list.png[View logs in the {app-name}, width=40%,pdfwidth=40%]
+
+To view log details or search inside a log, tap on a log from the log list. Three things can happen:
```
```suggestion
To view details or search inside a log file, tap on the respective entry in the log list. Three things can happen:
```

EParzefall commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -37,11 +44,41 @@ image:settings/allow-light-filtering-apps.png[Allow light filtering apps, width=
+image:logs/no-log-reading-app-found.png[No log reading app found, width=40%,pdfwidth=40%]
 
-If you detect a problem or a bug in the {app-name}, you can send us logs about the problem. You will find the log information in the Settings view, after enabling it. To enable the logs, you have to tap five times on the version and build information, available under menu:Settings[ownCloud for Android] .
+. Exact one log reading app is installed
```
```suggestion
. Exactly one log reading app is installed
```

phil-davis replied:
```suggestion
. Exactly one log reading app is installed
```

EParzefall reviewed (approved):
Some language issues.

phil-davis commented on modules/ROOT/pages/settings.adoc:
```diff
@@ -37,11 +44,41 @@ image:settings/allow-light-filtering-apps.png[Allow light filtering apps, width=
-When enabled, inside the log section options to send and delete history are available as well as to search the logs. If you tap btn:[Send History] your mail app is opened to address the log file. Searching the logs is available via the lens icon in the top right-hand side of the Files view.
+. Select a log reading app from the list of available ones
++
+After selecting the app, the log gets opened by the app selected.
```
```suggestion
After selecting the app, the log is opened by the app selected.
```

mmattel commented on modules/ROOT/pages/connecting.adoc:
```diff
@@ -1,15 +1,27 @@
 
+== Physical Location of User Files
+
+Compared to earlier versions of this app (< 2.19), the location for files created or managed by this app is the so called *Scoped Storage*. Scoped Storage is, compared to Shared Storage, protected and part of the app. It is located on the devices memory and limited by its capacity. Files are now safe and private and can only be accessed by other apps that use the {document-provider-url}[Documents ...
```
According to: https://developer.android.com/topic/performance/memory-management#types_of_memory
it is **storage** or **persistant memory**
I will adopt the text accordingly.

mmattel commented:
Bad, there is no "Add suggestion to batch" tab in files....?

mmattel commented:
I have added all open suggestable commits which were in conflict because of double editing here in the PR in my local dev env, squashed and force pushed.

@phil-davis I also fixed the memory - storage/persistent comment you raised.
